central/networkgraph/entity/datastore/datastore_impl.go

```diff
@@ -175,6 +175,10 @@ func (ds *dataStoreImpl) GetAllEntitiesForCluster(ctx context.Context, clusterID
 		if entity.GetScope().GetClusterId() != "" && entity.GetScope().GetClusterId() != clusterID {
 			return false
 		}
+		// By definition, learned entities are discovered by the cluster. So we don't want to send them.
+		if entity.GetInfo().GetExternalSource().GetLearned() {
+			return false
+		}
```

Comment:
both of these are obvious query parameters to filter as much as possible out in the database.

Reply:
This is absolutely true. We would like to replace as much walkers as possible in a following PR (and now we know how to achieve it :tada: ). https://issues.redhat.com/browse/ROX-25850